code/modules/admin/topic.dm

```diff
@@ -349,6 +349,9 @@
 		if(!check_rights(R_SERVER))
 			return
 
+		GLOB.admin_delay_notice = input(usr, "Enter a reason for delaying the round end", "Round Delay Reason")
+		if(!GLOB.admin_delay_notice)
+			GLOB.admin_delay_notice = "An admin has already delayed the round end."

```

Comment:
This part, when combined with the part below, would act odd.

If you don't give a reason, it would say to the restarting admin: `An admin has already delayed the round end for the following reason: An admin has already delayed the round end.`

It should also say the message on the check antag screen next to the delay restart/undelay restart button

It should also remove this message if you undelay the round. right now it would keep it and show it to the restarting admin, even if the round end is no longer delayed. that seems silly..

The round end delay reason prompt should also have a cancel button, you can do this by adding `as null|text` to the end of the input line, so it becomes `GLOB.admin_delay_notice = input(usr, "Enter a reason for delaying the round end", "Round Delay Reason") as null|text`

Be sure to check `null` and `""` separately. `""` will be what shows up if they leave it blank, null will be what happens if they hit cancel. if they hit cancel it should return at this point.

it should only require the reason. if they are delaying the round, this bit of code also gets called if they undelay the round by pressing the button again. so you should check `SSticker.delay_end`